modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java

```diff
@@ -298,12 +300,19 @@ public String getSchemaType(Schema p) {
 
         if (ref != null && !ref.isEmpty()) {
             type = openAPIType;
-        } else if (typeMapping.containsKey(openAPIType)) {
-            type = typeMapping.get(openAPIType);
-            if (languageSpecificPrimitives.contains(type))
-                return (type);
-        } else
-            type = openAPIType;
+        } else {
+            // Handle "any type" as an empty interface
+            if (openAPIType.equals("object") && p != null && !ModelUtils.isObjectSchema(p) && !ModelUtils.isMapSchema(p)) {
```

Comment:
Hmm, it seems this would only work with the following suggestion.

```suggestion
            if (openAPIType.equals("object") && p != null && ModelUtils.isObjectSchema(p) && !ModelUtils.isMapSchema(p)) {
```

 Indeed, a schema like the following will be an `ObjectSchema`:
```yaml
AnyType:
  type: object
```